crates/ruff_diagnostics/src/diagnostic.rs

```diff
@@ -42,21 +42,22 @@ impl Diagnostic {
         }
     }
 
-    pub fn amend(&mut self, fix: Edit) -> &mut Self {
+    /// Set the location of the diagnostic's parent node.
+    pub fn set_parent(&mut self, parent: Location) {
```

Comment:
Alternatively, we could use `with_parent` and continue to return `self`, but we don't use that behavior anywhere.